### DISCUSSION TITLE
Add the ability to "Disable object picking" on conditions

### DISCUSSION THREAD

**Discussion by tristanbob** [4ian/GDevelop#3771] [Enhancement ideas (and their technical discussions)]
Currently, any condition that includes an object will perform two distinct functions:

```
1. TRUE/FALSE logic that determines if the actions and subevents are processed.
2. Performs object picking, which limits which instances will be affected by the actions.
```

There are some use-cases where the developer only wants the TRUE/FALSE feature of conditions; they do NOT want object picking to be performed.  For example, in my Colony Crawler game, I need to specify which objects get physics joints applied to them.   

```
ForEach(ObjectA):
     Conditions:
     - ObjectA.Point("Weld1") is inside ObjectB 
     - NOT [ INVERT( ObjectA.Point("Weld1") is inside ObjectC ] (this is TRUE when zero instances match, but it also results in zero picked instances) 
     Actions:
     - Create weld joint between ObjectA and ObjectB
```

The first condition works great, it picks the right objects.  
For the second condition, I ONLY want to know the truth of the condition.  With object picking enabled, it removes all instances of ObjectA.

**Suggestion:**

- Add a boolean value on all object conditions to "Disable object picking"
- Add right-click action on conditions "Disable object picking" that works similar to "Invert condition"
- Display this status with a small icon similar to the "Invert condition"

**Comment by arthuro555** [comment]
I'd suggest a condition with subconditions, like the "not" condition, if we were to add this.

Note that currently, we have something designed for that use case: the "pick all object" allows resetting object picking for an object, so that you can execute the action on all instances or add some more conditions to target the objects you want.

**Reply by Silver-Streak** [reply]
I think we're still missing what Tristan is looking for, which would still likely fall under the "pick none" we've asked about/requested in the more object picking discussion/issues.

**Reply by tristanbob** [reply]
I don't think I am looking for "Pick none", which I assume removes all picked instances. 

I want a condition that does NOT change the currently picked instances.

**Reply by 4ian** [reply]
I guess we could have a special condition (like Or/Not/And) that does "Run the following conditions and forget picked objects:". This condition would be a "AND" (it's true if all sub conditions are true).

**Comment by Bouh** [comment]
A simple use case is that creating objects of an external layout does not allow selecting the instances just created.

To act on the newly created objects it is currently impossible to do so.

The easiest way would be to add an option to the action.
And an action that clears the current selection of objects.

What is expected for more flexibility is:
- Clear all picked instances (action or condition)
- In the action that creates the objects, a boolean: Pick instance created?
